src/adapter-types/src/dyncfgs.rs

```diff
@@ -66,7 +66,7 @@ pub const WITH_0DT_CAUGHT_UP_CHECK_CUTOFF: Config<Duration> = Config::new(
 /// Enable logging of statement lifecycle events in mz_internal.mz_statement_lifecycle_history.
 pub const ENABLE_STATEMENT_LIFECYCLE_LOGGING: Config<bool> = Config::new(
     "enable_statement_lifecycle_logging",
-    false,
+    true,
```

Comment:
This might be expensive, but we are also doing it in production.

Reply:
👍🏽 this seems very much worth doing. @ParkMyCar, want to weigh in on this though?